packages/eslint-plugin/__tests__/stylex-margin-padding.js

```diff
@@ -0,0 +1,262 @@
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ */
+
+'use strict';
+
+jest.disableAutomock();
+
+const { RuleTester: ESLintTester } = require('eslint');
+const rule = require('../src/stylex-margin-padding-shorthand');
+
+const eslintTester = new ESLintTester({
+  parser: require.resolve('hermes-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+eslintTester.run('stylex-sort-keys', rule.default, {
+  valid: [
+    {
+      code: `
+      import stylex from 'stylex';
+      const styles = stylex.create({
+        main: {
+          marginInlineEnd: '14px',
+          marginInlineStart: '14px',
+        },
+      })
+    `,
+    },
+  ],
+  invalid: [
+    {
+      code: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            margin: '10px 12px 13px 14px',
+          },
+        });
+      `,
+      output: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            marginInlineTop: '10px',
+            marginInlineEnd: '12px',
+            marginInlineBottom: '13px',
+            marginInlineStart: '14px',
+          },
+        });
+      `,
+      errors: [
+        {
+          message:
+            'Property shorthands like margin: 0 0 0 0 are not supported in styleX. Separate into marginStart, marginEnd [...].',
+        },
+      ],
+    },
+    {
+      code: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            margin: 10,
+          },
+        });
+      `,
+      output: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            marginInlineTop: 10,
+            marginInlineEnd: 10,
+            marginInlineBottom: 10,
+            marginInlineStart: 10,
+          },
+        });
+      `,
+      errors: [
+        {
+          message:
+            'Property shorthands like margin: 0 0 0 0 are not supported in styleX. Separate into marginStart, marginEnd [...].',
+        },
+      ],
+    },
+    {
+      code: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            margin: '10em 1em',
+          },
+        });
+      `,
+      output: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            marginInlineTop: '10em',
+            marginInlineEnd: '1em',
+            marginInlineBottom: '10em',
+            marginInlineStart: '1em',
+          },
+        });
+      `,
+      errors: [
+        {
+          message:
+            'Property shorthands like margin: 0 0 0 0 are not supported in styleX. Separate into marginStart, marginEnd [...].',
+        },
+      ],
+    },
+    {
+      code: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            margin: '10em 1em !important',
```

Comment:
I don't think we should handle values with !important, as we would aim to remove that from the StyleX API. We have some internal legacy use, but should probably only enable that for Meta behind a feature flag in the compiler. For XPR purposes we require Web code to get rid of any !important use anyway

Reply:
The idea was add this to the implementation for legacy code within Meta only. @mellyeliu is it possible to make this configurable with a ESLint config option?

Reply:
this is the approach I've taken - I added an `allowImportant` config option defaulted to false. When true, the fixer preserves !important tags in the resultant code. When false, the fixer strips those values.